snap/info.go

```diff
@@ -409,6 +409,12 @@ type SlotInfo struct {
 	Apps      map[string]*AppInfo
 }
 
+// SocketInfo provides information on application sockets.
+type SocketInfo struct {
```

Comment:
The syntax used is a bit unclear to me. What does it mean to have multiple socket names pointing to such a structure, when for each socket file we need a corresponding service file, and we only have a single service file for each application listed?

Also, we can have multiple ListenStreams for a single socket file, which apparently conflicts with this schema?

I'll ignore this issue for the rest of the review, but we need to understand how these things will work before moving on.

Reply:
You don't need to have a .service file for each .socket file, if the .socket defines `Service=<service name>`. This allows having multipel sockets activating one service and controlling them individually.

I'm not sure if you can define multiple `ListenStream`s in a single file, but even if it is having them separate gives more control (for instance the service itself could disable some sockets under certain conditions)

Reply:
Yes, multiple ListenStreams are supported, and snapd itself uses that.

That said, you are right. Allowing multiple files makes it more flexible, so +1.